core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/IntelliJResolver.scala

```diff
@@ -37,9 +43,44 @@ case class IntelliJPatternResolver(pattern: String) extends IntelliJResolver {
       "version" -> version.releaseOrBuild,
       "release" -> version.release.getOrElse("snapshot-release")
     )
-    val replaced = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
+    val replacedBeforeResolvingGlobs = replacements.foldLeft(pattern) { case (path, (pattern, replacement)) =>
       path.replace(s"[$pattern]", replacement)
     }
+    val replaced = replaceGlobsWithExistingDirectories(List(replacedBeforeResolvingGlobs)).head
     Dependency(replaced)
   }
+
+  // solution below assumes that each * character is used to mark one part of the path (one atomic directory),
+  // for example: "file:///home/.cache/ides/com.jetbrains.intellij.idea/ideaIC/[revision]/*/ideaIC-[revision]/"
+  @tailrec
+  private def replaceGlobsWithExistingDirectories(paths: List[String]): List[String] =
+    if (paths.exists(pathMightBeValidResource))
+      paths.filter(pathMightBeValidResource)
+    else
+      replaceGlobsWithExistingDirectories(paths.flatMap(replaceFirstFoundWildcardWithDirectories))
+
+  private def pathMightBeValidResource(path: String): Boolean =
+    path.indexOf('*') == -1 &&
+      (path.endsWith(".zip") ||
```

Comment:
isn't it too specific check? Filling extension is responsibility of pattern replacement that already happened.

Reply:
Yes, the check is needed to find proper path from multiple directories that exist in place of `*` (some are not valid).
But I just realised that this check is not valid. This check should return `true` only when `path` ends properly, as defined in the pattern... Let me fix this.

Reply:
Ok, logic of `pathMightBeValidResource` method is fixed now.